kad-dht/README.md

```diff
@@ -455,6 +455,53 @@ multiaddrs are stored in the node's peerbook.
 
 ---
 
+## Practical Examples
+<details>
+  <summary>Searching the <b>IPFS</b> Kad DHT for a Peer's Public Key</summary><blockquote>
+  
+  &nbsp;
+  ##### Lets find the Public Key for the IPFS Bootstrap node with the PeerID "QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+  1. Derive the key to search the DHT for
+     1. We're searching for a public key record therefore we're going to use the `/pk/` namespace
+     2. And we want the public key associated with the PeerID `QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`
+     3. Putting these together we get `/pk/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`
+     4. But we need to turn this into it's raw byte representation
+     5. For the namespace and forward slashes we just convert the utf8 string into bytes
+     6. For the PeerID we need to remember that this is a [base58btc encoding](peer-ids.md#decoding) 
```

Comment:
I think the example of en-/ and decoding of a peer ID would be better placed in https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md#decoding. We have some examples there, but none of them go all the way from private and public key to peer ID back to public key (in the case of ed25519). Would you be interested in extending the documentation there?